src/Lean/Server/FileWorker.lean

```diff
@@ -428,6 +428,7 @@ section MainLoop
     let state ← get
     let doc := state.doc
     doc.cancelTk.set
+    -- this is acting like an acknowledgement back to the Watchdog that exit was in fact received.
     publishDiagnostics doc.meta #[] ctx.hOut
```

Comment:
This still looks wrong, there is no guarantee that all asynchronously executing, possibly diagnostics-producing tasks have finished at this point - after all, we have flagged them for cancellation only in the line above.

Reply:
It works because the WorkerState in the watch dog has been set to “Exiting” but we just need this one more guaranteed msg from the worker to ensure we actually enter the forwardMessages loop one last time and it is there that we actually send the clear diagnostics as the very last thing and I’ve verified this is working as designed.  The problem is not when too many messages are sent from from the worker, the problem was when the worker sends none at all.

Reply:
Ok, but why introduce a fake message when we already have the worker process exit "message" at the location I linked to above?

Reply:
Oh, I see what you mean now, keep running the loop until the exception occurs with exit code 0 means normal exit, so that's a good time to clear the diagnostics, no need for a final message.  Ok, I changed it to this simpler version.

Reply:
Here's the updated diagram:

![image](https://user-images.githubusercontent.com/18707114/194669591-d9c1f3b5-c7d6-42cc-a0d8-891df4962872.png)